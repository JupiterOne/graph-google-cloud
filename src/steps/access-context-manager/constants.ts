export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES =
  'fetch-access-context-manager-access-policies';
export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS =
  'fetch-access-context-manager-access-levels';
export const STEP_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS =
  'fetch-access-context-manager-service-perimeters';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY = 'AccessPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY =
  'google_access_context_manager_access_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL = 'Ruleset';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL =
  'google_access_context_manager_access_level';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'google_access_context_manager_service_perimeter';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_EGRESS_POLICY =
  'ControlPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_EGRESS_POLICY =
  'google_access_context_manager_service_perimeter_egress_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_INGRESS_POLICY =
  'ControlPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_INGRESS_POLICY =
  'google_access_context_manager_service_perimeter_ingress_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_API_OPERATION =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_API_OPERATION =
  'google_access_context_manager_service_perimeter_api_operation';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_METHOD_SELECTOR =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_METHOD_SELECTOR =
  'google_access_context_manager_service_perimeter_method_selector';

export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_ACCESS_LEVEL =
  'google_access_context_manager_access_policy_has_level';
export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_SERVICE_PERIMETER =
  'google_access_context_manager_access_policy_has_service_perimeter';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_PROTECTS_PROJECT =
  'google_access_context_manager_service_perimeter_protects_project';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_LIMITS_API_SERVICES =
  'google_access_context_manager_service_perimeter_limits_api_service';

export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_HAS_EGRESS_POLICY =
  'google_access_context_manager_service_perimeter_has_egress_policy';
export const RELATIONSHIP_TYPE_EGRESS_POLICY_HAS_API_OPERATION =
  'google_access_context_manager_service_perimeter_egress_policy_has_api_operation';
export const RELATIONSHIP_TYPE_API_OPERATION_HAS_METHOD_SELECTOR =
  'google_access_context_manager_service_perimeter_api_operation_has_method_selector';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_HAS_INGRESS_POLICY =
  'google_access_context_manager_service_perimeter_has_ingress_policy';
export const RELATIONSHIP_TYPE_INGRESS_POLICY_HAS_API_OPERATION =
  'google_access_context_manager_service_perimeter_ingress_policy_has_api_operation';

export const IngestionSources = {
  ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES: 'acm-access-policies',
  ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS: 'acm-access-levels',
  ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS: 'acm-service-perimeters',
};

export const AccessContextManagerIngestionConfig = {
  [IngestionSources.ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES]: {
    title: 'Google Cloud Access Context Manager Access Policies',
    description: 'Defines secure access boundaries in GCP.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS]: {
    title: 'Google Cloud Access Context Manager Access Levels',
    description: 'Manages access via hierarchical levels in GCP.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS]: {
    title: 'Google Cloud Access Context Manager Service Perimeters',
    description: 'Secures resources within network boundaries.',
    defaultsToDisabled: false,
  },
};
