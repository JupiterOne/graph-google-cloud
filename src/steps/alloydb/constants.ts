export const STEP_ALLOYDB_POSTGRE_SQL_SERVICE = 'create-postgre-sql-service';
export const STEP_ALLOYDB_POSTGRE_SQL_CLUSTER =
  'fetch-alloydb-postgre-sql-cluster';
export const STEP_ALLOYDB_POSTGRE_SQL_INSTANCE =
  'fetch-alloydb-postgre-sql-instance';
export const STEP_ALLOYDB_POSTGRE_SQL_CONNECTION =
  'fetch-alloydb-postgre-sql-connection';
export const STEP_ALLOYDB_POSTGRE_SQL_BACKUP =
  'fetch-alloydb-postgre-sql-backup';

// Entity type
export const ENTITY_TYPE_ALLOYDB_POSTGRE_SQL_SERVICE = 'google_cloud_alloydb';
export const ENTITY_CLASS_ALLOYDB_POSTGRE_SQL_SERVICE = ['Service'];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_CLUSTER =
  'google_cloud_alloydb_cluster';
export const ENTITY_CLASS_POSTGRE_SQL_CLUSTER = [
  'Database',
  'DataStore',
  'Cluster',
];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_INSTANCE =
  'google_cloud_alloydb_instance';
export const ENTITY_CLASS_POSTGRE_SQL_INSTANCE = [
  'Database',
  'DataStore',
  'Host',
];

export const ENTITY_TYPE_POSTGRE_SQL_CONNECTION =
  'google_cloud_alloydb_connection';
export const ENTITY_CLASS_POSTGRE_SQL_CONNECTION = ['Network'];

export const ENTITY_TYPE_POSTGRE_SQL_BACKUP = 'google_cloud_alloydb_backup';
export const ENTITY_CLASS_POSTGRE_SQL_BACKUP = ['Backup'];

// Relationships
export const STEP_PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP =
  'build-project-alloydb-service-relationship';
export const STEP_PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-project-has-alloydb-cluster-relationship';

export const STEP_ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP =
  'build-alloydb-instance-uses-cluster-relationship';

export const STEP_ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP =
  'build-alloydb-cluster-has-backup-relationship';

export const STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP =
  'build-alloydb-instance-has-connection-relationship';

export const STEP_ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP =
  'build-alloydb-cluster-uses-kms-key-relationship';

export const STEP_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-user-asssigned-alloydb-cluster-relatipnship';

export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_SERVICE =
  'google_cloud_project_has_alloydb';
export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_CLUSTER =
  'google_cloud_project_has_alloydb_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_INSTANCE_USES_CLUSTER =
  'google_cloud_alloydb_instance_uses_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_HAS_BACKUP =
  'google_cloud_alloydb_cluster_has_backup';
export const RELATIONSHIP_TYPE_STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_BACKUP =
  'google_cloud_alloydb_instance_has_connection';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_USES_KMS_KEY =
  'google_cloud_alloydb_cluster_uses_kms_crypto_key';
export const RELATIONSHIP_TYPE_USER_ASSIGNED_ALLOYDB_CLUSTER =
  'google_user_assigned_cloud_alloydb_cluster';

// IngestionSources
export const IngestionSources = {
  ALLOYDB_POSTGRE_SQL_SERVICE: 'postgre-sql-service',
  ALLOYDB_POSTGRE_SQL_CLUSTER: 'alloydb-postgre-sql-cluster',
  ALLOYDB_POSTGRE_SQL_INSTANCE: 'alloydb-postgre-sql-instance',
  ALLOYDB_POSTGRE_SQL_CONNECTION: 'alloydb-postgre-sql-connection',
  ALLOYDB_POSTGRE_SQL_BACKUP: 'alloydb-postgre-sql-backup',
  PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP:
    'project-has-alloydb-service-relationship',
  PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP:
    'project-has-alloydb-cluster-relationship',
  ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP:
    'alloydb-instance-uses-cluster-relationship',
  ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP:
    'alloydb-cluster-has-backup-relationship',
  ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP:
    'alloydb-instance-has-connection-relationship',
  ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP:
    'alloydb-cluster-uses-kms-key-relationship',
  GOOGLE_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP:
    'google-user-assigned-alloydb-cluster-relationship',
};

// IngestionSources Configs
export const AlloyDBIngestionConfig = {
  [IngestionSources.ALLOYDB_POSTGRE_SQL_SERVICE]: {
    title: 'AlloyDB for Postgre SQL service',
    description: 'AlloyDB for Postgre SQL service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CLUSTER]: {
    title: 'AlloyDB for Postgre SQL Cluster',
    description: 'AlloyDB for Postgre SQL Cluster.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_INSTANCE]: {
    title: 'AlloyDB for Postgre SQL Instance',
    description: 'AlloyDB for Postgre SQL Instance.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CONNECTION]: {
    title: 'AlloyDB for Postgre SQL Connection',
    description: 'AlloyDB for Postgre SQL Connection.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP]: {
    title: 'Project AlloyDB Service Has Relationship',
    description: 'Build Project AlloyDB Service Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP]: {
    title: 'Project AlloyDB Cluster Has Relationship',
    description: 'Build Project AlloyDB Cluster Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP]: {
    title: 'AlloyDB Instance Cluster Uses Relationship',
    description: 'Build AlloyDB Instance Cluster Uses Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP]: {
    title: 'AlloyDB Instance Backup Has Relationship',
    description: 'Build AlloyDB Backup Cluster Has  Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP]: {
    title: 'AlloyDB Instance Connection Has  Relationship',
    description: 'Build AlloyDB Instance Connection Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP]: {
    title: 'AlloyDB Cluster Kms Key Uses Relationship',
    description: 'Build AlloyDB Cluster Kms Key Has Relationship.',
    defaultsToDisabled: false,
  },
};

// IAM Permissions
export const AlloyDBPermissions = {
  STEP_ALLOYDB_POSTGRE_SQL_CLUSTER: ['alloydb.clusters.list'],
  STEP_ALLOYDB_POSTGRE_SQL_INSTANCE: ['alloydb.instances.list'],
  STEP_ALLOYDB_POSTGRE_SQL_CONNECTION: ['alloydb.instances.connect'],
  STEP_ALLOYDB_POSTGRE_SQL_BACKUP: ['alloydb.backups.get'],
};
