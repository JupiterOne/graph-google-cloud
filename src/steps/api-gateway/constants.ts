export const STEP_API_GATEWAY_APIS = 'fetch-api-gateway-apis';
export const STEP_API_GATEWAY_API_CONFIGS = 'fetch-api-gateway-api-configs';
export const STEP_API_GATEWAY_GATEWAYS = 'fetch-api-gateway-gateways';

export const ENTITY_CLASS_API_GATEWAY_API = 'Service';
export const ENTITY_TYPE_API_GATEWAY_API = 'google_api_gateway_api';

export const ENTITY_CLASS_API_GATEWAY_API_CONFIG = 'Configuration';
export const ENTITY_TYPE_API_GATEWAY_API_CONFIG =
  'google_api_gateway_api_config';

export const ENTITY_CLASS_API_GATEWAY_GATEWAY = 'Gateway';
export const ENTITY_TYPE_API_GATEWAY_GATEWAY = 'google_api_gateway_gateway';

export const RELATIONSHIP_TYPE_API_GATEWAY_API_USES_CONFIG =
  'google_api_gateway_api_uses_config';
export const RELATIONSHIP_TYPE_API_GATEWAY_API_HAS_GATEWAY =
  'google_api_gateway_api_has_gateway';
export const RELATIONSHIP_TYPE_API_GATEWAY_API_CONFIG_USES_SERVICE_ACCOUNT =
  'google_api_gateway_api_config_uses_iam_service_account';

export const IngestionSources = {
  API_GATEWAY_APIS: 'api-gateway-apis',
  API_GATEWAY_API_CONFIGS: 'api-gateway-api-configs',
  API_GATEWAY_GATEWAYS: 'api-gateway-gateways',
};

export const ApiGatewayIngestionConfig = {
  [IngestionSources.API_GATEWAY_APIS]: {
    title: 'Google Cloud API Gateway APIs',
    description: 'Endpoint management for API gateways.',
    defaultsToDisabled: false,
  },
  [IngestionSources.API_GATEWAY_API_CONFIGS]: {
    title: 'Google Cloud API Gateway API Configurations',
    description: 'Config settings for API interfaces.',
    defaultsToDisabled: false,
  },
  [IngestionSources.API_GATEWAY_GATEWAYS]: {
    title: 'Google Cloud API Gateway Gateways',
    description: 'Networking gateways for API management.',
    defaultsToDisabled: false,
  },
};

export const ApiGatewayPermissions = {
  STEP_API_GATEWAY_APIS: [
    'apigateway.apis.getIamPolicy',
    'apigateway.apis.list',
  ],
  STEP_API_GATEWAY_API_CONFIGS: [
    'apigateway.apiconfigs.list',
    'apigateway.apiconfigs.getIamPolicy',
  ],
  STEP_API_GATEWAY_GATEWAYS: [
    'apigateway.gateways.list',
    'apigateway.gateways.getIamPolicy',
  ],
};
