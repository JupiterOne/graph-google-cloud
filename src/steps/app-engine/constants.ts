export const STEP_APP_ENGINE_APPLICATION = 'fetch-app-engine-application';
export const STEP_CREATE_APP_ENGINE_BUCKET_RELATIONSHIPS =
  'build-app-engine-application-bucket-relationships';
export const STEP_APP_ENGINE_SERVICES = 'fetch-app-engine-services';
export const STEP_APP_ENGINE_VERSIONS = 'fetch-app-engine-versions';
export const STEP_APP_ENGINE_INSTANCES = 'fetch-app-engine-instances';

export const ENTITY_TYPE_APP_ENGINE_APPLICATION =
  'google_app_engine_application';
export const ENTITY_CLASS_APP_ENGINE_APPLICATION = 'Application';

export const ENTITY_TYPE_APP_ENGINE_SERVICE = 'google_app_engine_service';
export const ENTITY_CLASS_APP_ENGINE_SERVICE = 'Container';

export const ENTITY_TYPE_APP_ENGINE_VERSION = 'google_app_engine_version';
export const ENTITY_CLASS_APP_ENGINE_VERSION = 'Service';

export const ENTITY_TYPE_APP_ENGINE_INSTANCE = 'google_app_engine_instance';
export const ENTITY_CLASS_APP_ENGINE_INSTANCE = 'Host';

export const RELATIONSHIP_TYPE_APP_ENGINE_APPLICATION_HAS_SERVICE =
  'google_app_engine_application_has_service';

export const RELATIONSHIP_TYPE_APP_ENGINE_SERVICE_HAS_VERSION =
  'google_app_engine_service_has_version';

export const RELATIONSHIP_TYPE_APP_ENGINE_VERSION_HAS_INSTANCE =
  'google_app_engine_version_has_instance';

export const RELATIONSHIP_TYPE_APP_ENGINE_APPLICATION_USES_BUCKET =
  'google_app_engine_application_uses_storage_bucket';

export const RELATIONSHIP_TYPE_GOOGLE_USER_CREATED_VERSION =
  'mapping_source_created_google_user';

export const RELATIONSHIP_TYPE_SERVICE_ACCOUNT_CREATED_VERSION =
  'google_iam_service_account_created_app_engine_version';

export const IngestionSources = {
  APP_ENGINE_APPLICATION: 'app-engine-application',
  APP_ENGINE_SERVICES: 'app-engine-services',
  APP_ENGINE_VERSIONS: 'app-engine-versions',
  APP_ENGINE_INSTANCES: 'app-engine-instances',
};

export const AppEngineIngestionConfig = {
  [IngestionSources.APP_ENGINE_APPLICATION]: {
    title: 'Google Cloud App Engine Application',
    description: 'Platform for building scalable web apps.',
    defaultsToDisabled: false,
  },
  [IngestionSources.APP_ENGINE_SERVICES]: {
    title: 'Google Cloud App Engine Services',
    description: 'Modular components of App Engine apps.',
    defaultsToDisabled: false,
  },
  [IngestionSources.APP_ENGINE_VERSIONS]: {
    title: 'Google Cloud App Engine Versions',
    description: 'Versioning for App Engine application components.',
    defaultsToDisabled: false,
  },
  [IngestionSources.APP_ENGINE_INSTANCES]: {
    title: 'Google Cloud App Engine Instances',
    description: 'Running instances of App Engine services.',
    defaultsToDisabled: false,
  },
};
