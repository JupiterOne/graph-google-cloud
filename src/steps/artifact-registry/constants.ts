export const ARTIFACT_REGISTRY_REPOSITORY_CLASS = ['CodeRepo', 'Repository'];
export const ARTIFACT_REGISTRY_REPOSITORY_TYPE =
  'google_cloud_artifact_registry_repository';
export const STEP_ARTIFACT_REGISTRY_REPOSITORY =
  'fetch-artifact-registry-repository';

export const ARTIFACT_REPOSITORY_PACKAGE_CLASS = ['CodeModule'];
export const ARTIFACT_REPOSITORY_PACKAGE_TYPE =
  'google_cloud_artifact_registry_package';
export const STEP_ARTIFACT_REPOSIOTRY_PACKAGE =
  'fetch-artifact-repository-package';

export const STEP_ARTIFACT_REGISTRY = 'fetch-artifact-registry';
export const ARTIFACT_REGISTRY_CLASS = ['Service'];
export const ARTIFACT_REGISTRY_TYPE = 'google_cloud_artifact_registry';

export const STEP_ARTIFACT_REGISTRY_VPCSC_CONFIGURATION =
  'fetch-artifact-registry-vpcsc-configuration';
export const ARTIFACT_REGISTRY_VPCSC_CONFIGURATION_CLASS = ['Configuration'];
export const ARTIFACT_REGISTRY_VPCSC_CONFIGURATION_TYPE =
  'google_cloud_artifact_registry_vpcsc_configuration';

export const STEP_PROJECT_HAS_ARTIFACT_REGISTRY_REPOSITORY_RELATIONSHIP =
  'fetch-project-has-artifact-registry-repository';
export const RELATIONSHIP_PROJECT_HAS_ARTIFACT_REGISTRY_REPOSITORY_TYPE =
  'google_cloud_project_has_artifact_registry_repository';

export const STEP_PROJECT_HAS_ARTIFACT_REGISTRY_RELATIONSHIP =
  'fetch-project-has-artifact-registry';
export const RELATIONSHIP_PROJECT_HAS_ARTIFACT_REGISTRY_TYPE =
  'google_cloud_project_has_artifact_registry';

export const STEP_ARTIFACT_REGISTRY_REPOSITORY_USES_KMS_KEY_RELATIONSHIP =
  'fetch-artifact-registry-repository-uses-kms-key';
export const RELATIONSHIP_ARTIFACT_REGISTRY_REPOSITORY_USES_KMS_KEY_TYPE =
  'google_cloud_artifact_registry_repository_uses_kms_crypto_key';

export const STEP_ARTIFACT_REGISTRY_REPOSITORY_USES_PACKAGE_RELATIONSHIP =
  'fetch-artifact-registry-repository-uses-package';
export const RELATIONSHIP_ARTIFACT_REGISTRY_REPOSITORY_USES_PACKAGE_TYPE =
  'google_cloud_artifact_registry_repository_uses_package';

export const STEP_ARTIFACT_REGISTRY_REPOSITORY_USES_NPM_PACKAGE_RELATIONSHIP =
  'fetch-artifact-registry-repository-uses-npm-package';
export const RELATIONSHIP_ARTIFACT_REGISTRY_REPOSITORY_USES_NPM_PACKAGE_TYPE =
  'google_cloud_artifact_registry_repository_uses_npm_package';

export const STEP_ARTIFACT_REPOSITROY_PACKAGE_IS_NPM_PACKAGE_RELATIONSHIP =
  'fetch-artifact-repository-package-is-npm-package';
export const RELATIONSHIP_ARTIFACT_REPOSITORY_PACKAGE_IS_NPM_PACKAGE_TYPE =
  'google_cloud_artifact_registry_package_is_npm_package';

export const IngestionSources = {
  ARTIFACT_REGISTRY_REPOSITORY: 'artifact_registry_repositories',
  ARTIFACT_REPOSITORY_PACKAGE: 'artifact_repository_packages',
  ARTIFACT_REGISTRY: 'artifact_registries',
  ARTIFACT_REGISTRY_VPCSC_CONFIGURATION:
    'artifact_registry_vpcsc_configurations',
  PROJECT_HAS_ARTIFACT_REGISTRY_REPOSITORY_RELATIONSHIP:
    'project_has_artifact_registry_repository_relationship',
  PROJECT_HAS_ARTIFACT_REGISTRY_RELATIONSHIP:
    'project_has_artifact_registry_relationship',
  ARTIFACT_REGISTRY_REPOSITORY_USES_KMS_KEY_RELATIONSHIP:
    'artifact_registry_repository_uses_kms_key_relationship',
  ARTIFACT_REGISTRY_REPOSITORY_USES_PACKAGE_RELATIONSHIP:
    'artifact_registry_repository_uses_package_relationship',
  ARTIFACT_REGISTRY_REPOSITORY_USES_NPM_PACKAGE_RELATIONSHIP:
    'artifact_registry_repository_uses_npm_package_relationship',
  ARTIFACT_REPOSITORY_PACKAGE_IS_NPM_PACKAGE_REALTIONSHIP:
    'artifact_repository_package_is_npm_package_relationship',
};

export const MonitoringIngestionConfig = {
  [IngestionSources.ARTIFACT_REGISTRY_REPOSITORY]: {
    title: 'Artifact Registry Repository',
    description: 'Artifact Registry Repository for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REGISTRY]: {
    title: 'Artifact Registry',
    description: 'Artifact Registry for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REPOSITORY_PACKAGE]: {
    title: 'Artifact Repository Package',
    description: 'Artifact Repository package for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REGISTRY_VPCSC_CONFIGURATION]: {
    title: 'Artifact Registry vpcsc configuration',
    description: 'Artifact Registry vpcsc configuration for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ARTIFACT_REGISTRY_REPOSITORY_RELATIONSHIP]: {
    title: 'Project HAS artifact registry repository',
    description: 'Project HAS artifact registry repository for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ARTIFACT_REGISTRY_RELATIONSHIP]: {
    title: 'Project HAS artifact registry repository',
    description: 'Project HAS artifact registry repository for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REGISTRY_REPOSITORY_USES_KMS_KEY_RELATIONSHIP]: {
    title: 'Artifact registry repository uses kms crypto key',
    description: 'Artifact registry repository uses kms crypto key for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REGISTRY_REPOSITORY_USES_PACKAGE_RELATIONSHIP]: {
    title: 'Artifact registry repository uses packages',
    description: 'Artifact registry repository uses packages for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.ARTIFACT_REGISTRY_REPOSITORY_USES_NPM_PACKAGE_RELATIONSHIP]:
    {
      title: 'Artifact registry repository uses npm package',
      description: 'Artifact registry repository uses npm package for GCP',
      defaultsToDisabled: false,
    },
  [IngestionSources.ARTIFACT_REPOSITORY_PACKAGE_IS_NPM_PACKAGE_REALTIONSHIP]: {
    title: 'Artifact repository package is npm package',
    description: 'Artifact repository package is npm package for GCP',
    defaultsToDisabled: false,
  },
};

//Fetched directly from the repository.location.list
// This is the location : https://cloud.google.com/artifact-registry/docs/repositories/repo-locations

export const ARTIFACT_REGISTRY_LOCATIONS: string[] = [
  'africa-south1',
  'asia',
  'asia-east1',
  'asia-east2',
  'asia-northeast1',
  'asia-northeast2',
  'asia-northeast3',
  'asia-south1',
  'asia-south2',
  'asia-southeast1',
  'asia-southeast2',
  'australia-southeast1',
  'australia-southeast2',
  'europe',
  'europe-central2',
  'europe-north1',
  'europe-southwest1',
  'europe-west1',
  'europe-west10',
  'europe-west12',
  'europe-west2',
  'europe-west3',
  'europe-west4',
  'europe-west6',
  'europe-west8',
  'europe-west9',
  'me-central1',
  'me-central2',
  'me-west1',
  'northamerica-northeast1',
  'northamerica-northeast2',
  'southamerica-east1',
  'southamerica-west1',
  'us',
  'us-central1',
  'us-east1',
  'us-east4',
  'us-east5',
  'us-south1',
  'us-west1',
  'us-west2',
  'us-west3',
  'us-west4',
];
