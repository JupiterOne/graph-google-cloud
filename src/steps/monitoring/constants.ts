export const STEP_MONITORING_ALERT_POLICIES = 'fetch-monitoring-alert-policies';

export const MONITORING_ALERT_POLICY_CLASS = 'Policy';
export const MONITORING_ALERT_POLICY_TYPE = 'google_monitoring_alert_policy';

export const IngestionSources = {
  MONITORING_ALERT_POLICIES: 'monitoring-alert-policies',
};

export const MonitoringIngestionConfig = {
  [IngestionSources.MONITORING_ALERT_POLICIES]: {
    title: 'Google Monitoring Alert Policies',
    description: 'Alert policies for GCP resources.',
    defaultsToDisabled: false,
  },
};
