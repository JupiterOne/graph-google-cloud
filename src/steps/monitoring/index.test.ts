import {
  executeStepWithDependencies,
  Recording,
  StepTestConfig,
} from '@jupiterone/integration-sdk-testing';
import { STEP_MONITORING_ALERT_POLICIES } from './constants';
import {
  setupGoogleCloudRecording,
  getMatchRequestsBy,
} from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { invocationConfig } from '../../index';

describe(STEP_MONITORING_ALERT_POLICIES, () => {
  let recording: Recording;
  afterEach(async () => {
    if (recording) await recording.stop();
  });

  jest.setTimeout(450000);

  test(STEP_MONITORING_ALERT_POLICIES, async () => {
    recording = setupGoogleCloudRecording({
      name: STEP_MONITORING_ALERT_POLICIES,
      directory: __dirname,
      options: {
        matchRequestsBy: getMatchRequestsBy(integrationConfig),
      },
    });

    const stepTestConfig: StepTestConfig = {
      stepId: STEP_MONITORING_ALERT_POLICIES,
      instanceConfig: integrationConfig,
      invocationConfig: invocationConfig as any,
    };

    const result = await executeStepWithDependencies(stepTestConfig);
    expect(result).toMatchStepMetadata(stepTestConfig);
  });
});
