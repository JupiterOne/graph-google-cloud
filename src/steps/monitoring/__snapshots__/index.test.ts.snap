// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAlertPolicies should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Policy",
      ],
      "_key": "projects/j1-gc-integration-dev-v2/alertPolicies/8900061597970753716",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "combiner": "OR",
            "conditions": [
              {
                "conditionThreshold": {
                  "aggregations": [
                    {
                      "alignmentPeriod": "300s",
                      "perSeriesAligner": "ALIGN_RATE",
                    },
                  ],
                  "comparison": "COMPARISON_GT",
                  "duration": "0s",
                  "filter": "metric.type="logging.googleapis.com/user/my-example-metric" AND resource.type="metric"",
                  "trigger": {
                    "count": 1,
                  },
                },
                "displayName": "test condition",
                "name": "projects/j1-gc-integration-dev-v2/alertPolicies/8900061597970753716/conditions/8900061597970755317",
              },
            ],
            "creationRecord": {
              "mutateTime": "2021-02-19T13:41:42.161511391Z",
              "mutatedBy": "j1-gc-integration-dev-v2-tf@j1-gc-integration-dev-v2.iam.gserviceaccount.com",
            },
            "displayName": "Example Alert Policy",
            "enabled": true,
            "mutationRecord": {
              "mutateTime": "2021-02-19T13:41:42.161511391Z",
              "mutatedBy": "j1-gc-integration-dev-v2-tf@j1-gc-integration-dev-v2.iam.gserviceaccount.com",
            },
            "name": "projects/j1-gc-integration-dev-v2/alertPolicies/8900061597970753716",
          },
        },
      ],
      "_type": "google_monitoring_alert_policy",
      "conditionFilters": [
        "metric.type="logging.googleapis.com/user/my-example-metric" AND resource.type="metric"",
      ],
      "content": "",
      "createdOn": 1613742102161,
      "displayName": "Example Alert Policy",
      "enabled": true,
      "name": "Example Alert Policy",
      "summary": "Alert policy that is triggered based on some metric",
      "title": "Alert policy",
      "updatedOn": 1613742102161,
      "webLink": "https://console.cloud.google.com/monitoring/alerting/policies/8900061597970753716?project=j1-gc-integration-dev-v2",
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_monitoring_alert_policy",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
